Mathlib/MeasureTheory/OuterMeasure/Caratheodory.lean

```diff
@@ -71,6 +71,17 @@ theorem isCaratheodory_union (h₁ : IsCaratheodory m s₁) (h₂ : IsCaratheodo
     union_diff_left, h₂ (t ∩ s₁)]
   simp [diff_eq, add_assoc]
 
+lemma isCaratheodory_iUnion_finite {ι : Type*} {s : ι → Set α} {t : Set ι} (ht : t.Finite)
```

Comment:
```suggestion
lemma IsCaratheodory.biUnion_of_finite {ι : Type*} {s : ι → Set α} {t : Set ι} (ht : t.Finite)
```

Reply:
This doesn't work, unfortunately. There's no hypothesis `IsCaratheodory foo bar` to dot-notate off, only `(h : ∀ i ∈ t, m.IsCaratheodory (s i))` which isn't dot-notateable. So I have gone with `isCaratheodory_biUnion_of_finite`.

Reply:
No, but there is anonymous dot notation available, cf https://github.com/leanprover-community/mathlib4/pull/20545#discussion_r1916007568

Reply:
Well, the suggestion you made did not work.

Reply:
Is it because the lemma here is in the wrong namespace, possibly?

Reply:
Sorry, I see now! Not sure what I was doing wrong before.